x-pack/plugins/index_management/server/routes/api/component_templates/delete.ts

```diff
@@ -14,40 +14,45 @@ const paramsSchema = schema.object({
   names: schema.string(),
 });
 
-export const registerDeleteRoute = ({ router }: RouteDependencies): void => {
+export const registerDeleteRoute = ({
+  router,
+  lib: { handleEsError },
+}: RouteDependencies): void => {
   router.delete(
     {
       path: addBasePath('/component_templates/{names}'),
       validate: {
         params: paramsSchema,
       },
     },
-    async (ctx, req, res) => {
-      const { callAsCurrentUser } = ctx.dataManagement!.client;
-      const { names } = req.params;
+    async (context, request, response) => {
+      const { client } = context.core.elasticsearch;
+      const { names } = request.params;
       const componentNames = names.split(',');
 
-      const response: { itemsDeleted: string[]; errors: any[] } = {
+      const responseBody: { itemsDeleted: string[]; errors: any[] } = {
         itemsDeleted: [],
         errors: [],
       };
 
       await Promise.all(
-        componentNames.map((componentName) => {
-          return callAsCurrentUser('dataManagement.deleteComponentTemplate', {
-            name: componentName,
-          })
-            .then(() => response.itemsDeleted.push(componentName))
-            .catch((e) =>
-              response.errors.push({
-                name: componentName,
-                error: e,
-              })
-            );
+        componentNames.map(async (componentName) => {
```

Comment:
I updated this code to match the pattern we're using elsewhere. It should be functionally the same though.